src/index.ts

```diff
@@ -459,60 +456,103 @@ async function explicitInit(config: IClientConfig): Promise<EppoClient> {
 
     // We have two at-bats for initialization: from the configuration store and from fetching
     // We can resolve the initialization promise as soon as either one succeeds
+
+    // The definition of a successful configuration load differs for a cached config vs a fetched config.
+    // Specifically, an cached config with no entries is not considered a completed load and will result in
+    // `ConfigLoaderStatus.DID_NOT_PRODUCE` while and empty config from the fetch is considered a valid, `COMPLETED`
+    //  configuration load.
     let initFromConfigStoreError = undefined;
     let initFromFetchError = undefined;
 
-    const attemptInitFromConfigStore = configurationStore
+    const attemptInitFromConfigStore: ConfigurationLoadAttempt = configurationStore
       .init()
       .then(async () => {
         if (!configurationStore.getKeys().length) {
           // Consider empty configuration stores invalid
           applicationLogger.warn('Eppo SDK cached configuration is empty');
           initFromConfigStoreError = new Error('Configuration store was empty');
-          return '';
+          return ConfigLoaderStatus.DID_NOT_PRODUCE;
         }
 
         const cacheIsExpired = await configurationStore.isExpired();
         if (cacheIsExpired && !config.useExpiredCache) {
           applicationLogger.warn('Eppo SDK set not to use expired cached configuration');
           initFromConfigStoreError = new Error('Configuration store was expired');
-          return '';
+          return ConfigLoaderStatus.DID_NOT_PRODUCE;
+        } else if (cacheIsExpired && config.useExpiredCache) {
+          applicationLogger.warn('Eppo SDK config.useExpiredCache is true; using expired cache');
         }
-        return 'config store';
+
+        return ConfigLoaderStatus.COMPLETED;
       })
       .catch((e) => {
         applicationLogger.warn(
           'Eppo SDK encountered an error initializing from the configuration store',
           e,
         );
         initFromConfigStoreError = e;
+        return ConfigLoaderStatus.FAILED;
+      })
+      .then((status) => {
+        return [ConfigSource.CONFIG_STORE, status];
       });
-    const attemptInitFromFetch = instance
+
+    const attemptInitFromFetch: ConfigurationLoadAttempt = instance
       .fetchFlagConfigurations()
-      .then(() => 'fetch')
+      .then(() => {
+        console.log(instance.getFlagConfigurations());
+        if (configurationStore.isInitialized()) {
+          // If fetch has won the race and the configStore is initialized, that means we have a successful load and the
+          // fetched data was used to init.
+          return ConfigLoaderStatus.COMPLETED;
+        } else {
+          // If the config store is not initialized and the fetch says it is done, that means the fetch did not set any
+          // values into the config store (ex:the cache is not expired or invalid config data was received).
```

Comment:
If the cache is not expired then it would be initialized wouldn't it? I think did not produce means it was invalid or empty